buildstockbatch/cloud/docker_base.py

```diff
@@ -461,3 +462,44 @@ def run_simulations(cls, cfg, job_id, jobs_d, sim_dir, fs, output_path):
                 shutil.rmtree(item)
             elif os.path.isfile(item):
                 os.remove(item)
+
+    def log_summary(self):
+        """
+        Log a summary of how many simulations succeeded, failed, or ended with other statuses.
+
+        Uses the `completed_status` column of the files in results_csvs/results_*.csv.gz.
+        """
+        fs = self.get_fs()
+        # Summary of simulation statuses across all upgrades
+        status_summary = {}
+        all_statuses = set()
+
+        results_csv_dir = f"{self.results_dir}/results_csvs/"
+        try:
+            results_files = fs.ls(results_csv_dir)
```

Comment:
can we add a check here in case the results_csv does not exist for a particular upgrade? We have seen this in the past in which the results csv does exist for some upgrades but not for all of them and I wanted to make sure this capture that as well. Thoughts?

Reply:
Sure, added. Now you get logs like this if some upgrades are missing completely:
```
Baseline     Success: 8        Fail: 0        Invalid: 0        
Upgrade 01   Success: 8        Fail: 0        Invalid: 0        
Upgrade 02   Success: 8        Fail: 0        Invalid: 0        
Upgrade 03   Success: 8        Fail: 0        Invalid: 0        
Upgrade 04   Success: 2        Fail: 0        Invalid: 6        
Upgrade 05   Success: 8        Fail: 0        Invalid: 0        
Upgrade 06   Success: 0        Fail: 0        Invalid: 8        
Upgrade 07   Success: 8        Fail: 0        Invalid: 0        
Upgrade 08   Success: 8        Fail: 0        Invalid: 0        
Upgrade 09   Success: 8        Fail: 0        Invalid: 0        
Upgrade 10   Success: 8        Fail: 0        Invalid: 0        
Upgrade 11   Success: 8        Fail: 0        Invalid: 0        
Upgrade 12   Success: 8        Fail: 0        Invalid: 0        
No results found for Upgrade 13
No results found for Upgrade 14
```

Reply:
Great, thank you!